lib/Transforms/Utils/Local.cpp

```diff
@@ -331,6 +332,10 @@ bool llvm::isInstructionTriviallyDead(Instruction *I,
     if (Constant *C = dyn_cast<Constant>(CI->getArgOperand(0)))
       return C->isNullValue() || isa<UndefValue>(C);
 
+  // HLSL change - don't force unused convergenet markers to stay
+  if (CallInst *CI = dyn_cast<CallInst>(I))
+    if (hlsl::IsConvergentMarker(CI)) return true;
```

Comment:
I don't think we want to depend on code defined under lib/HLSL here.  Maybe we need to move IsConvergentMarker to dxilutil?

Reply:
I wondered about that. The implementation is pretty simple and easy to move. The reason I left it where it is was because of other places in lib/Transforms that include HLSL headers. Though not in lib/Transforms/Utils specifically

Reply:
I guess it might depend on whether it requires the HLSL.lib specifically?